convert2rhel/utils.py

```diff
@@ -197,7 +198,11 @@ def run_cmd_in_pty(cmd, expect_script=(), print_cmd=True, print_output=True, col
     process = PexpectSpawnWithDimensions(
         cmd[0],
         cmd[1:],
-        env={"LC_ALL": i18n.SCREENSCRAPED_LOCALE, "LANG": i18n.SCREENSCRAPED_LOCALE},
+        env={
+            "LC_ALL": i18n.SCREENSCRAPED_LOCALE,
+            "LANG": i18n.SCREENSCRAPED_LOCALE,
+            "LANGUAGE": i18n.SCREENSCRAPED_LOCALE,
```

Comment:
Thank you for pointing this out I will look into this and make the proper adjustments to the unit test!

Reply:
Hey @bocekm is this something I would implement in the code and unit test or is this strictly for the integration test?

Reply:
This is just about updating the existing or creating a new integration test.

As of now we've got this tier1/destructive test, setting just the LANG env var: https://github.com/oamg/convert2rhel/blob/v1.0/tests/integration/tier1/set-locale/use_non_english_language.py. That tests the issue where LANG unexpectedly has a precedence over LC_ALL in subscription-manager: https://github.com/oamg/convert2rhel/pull/386#issuecomment-987584120.

However, we should expect that other tools follow POSIX where LC_ALL has a precedence over LANG. So I think we should have a test with setting LANG to a non-English locale.

And lastly, there's the LANGUAGE variable. It's not defined in [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap08.html#tag_08_02) but in [GNU](https://www.gnu.org/software/gettext/manual/html_node/Locale-Environment-Variables.html). It has a precedence over any other locale env var. I'd say that we should have a test with setting LANGUAGE to a non-English locale.

@abadger, @danmyway, I'm not sure whether it's "safe" setting LANGUAGE, LC_ALL, and LANG in one test. I.e. whether having one test per env var would not be capable of revealing more. For the sake of saving computational power/time, I'm leaning towards saying set them all in one test. WDYT?

Reply:
@Andrew-ang9 I believe @bocekm is talking about the integration tests that should be added to make it more robust. It's not regarding your code/unit tests.

Reply:
@bocekm I agree, it is safe to set all of them in one test. The reason just a single test is fine is that the failure case we're looking for is when a program that we call returns translated strings instead of the strings from the `C` locale.  It's enough for us to set all of the environment variables that might cause this in a single test as any of them causing the program to return translated strings will cause the test to fail.  We might also want to set `LC_MESSAGES` to a non-English locale (This variable should not take precedence over `LC_ALL` or `LANGUAGE` but it also affects translations so it is probably good to make sure no programs are getting the precedence order incorrect).

Reply:
Also, the tests seem to be of tier1, so it's a destructive test. I think it's fine to set them without taking care of restoring it to original.